docs/hello_nextflow/06_hello_config.md

```diff
@@ -259,71 +515,617 @@ NXF_CHDIR=/home/gitpod/work/34/850fe31af0eb62a0eb1643ed77b84f
+    Just keep in mind that if you combine profiles with conflicting settings for the same directives, you might be surprised by the results.
+
+### Takeaway
+
+You now know how to change the executor and combine that with other environment settings using profiles.
+
+### What's next?
+
+Learn how to control the resources allocated for executing processes.
+
+---
+
+## 4. Allocate compute resources with process directives
+
+We've covered how to control what compute environment Nextflow is going to use to run the workflow, so now the next logical question is, how do we control the resources (CPU, memory etc) that will be allocated?
+
+The answer may not surprise you; it's process directives again.
+
+### 4.1. Increase default process resource allocations
+
+By default, Nextflow will use a single CPU and 2GB of memory for each process.
+Let's say we decide to double that.
+
+We can modify this behavior by setting the `cpu` and `memory` directives in the `process` block. Add the following to the end of your `nextflow.config` file:
+
+```groovy title="nextflow.config" linenums="20"
+process {
+    // defaults for all processes
+    cpus = 2
+    memory = 4.GB
+}
+```
+
+### 4.2. Run the workflow with the increased defaults
+
+Let's try that out, bearing in mind that we need to keep `-profile my_laptop` in the command going forward.
+
+```bash
+nextflow run main.nf -profile my_laptop
+```
+
+You may not notice any real difference in how quickly this runs, since this is such a small workload.
+But if you have a machine with few CPUs and you allocate a high number per process, you might see process calls getting queued behind each other.
+This is because Nextflow will ensure we aren't using more CPUs than are available.
+
+!!! tip
+
+    You can check the number of CPUs allocated to a given process by looking at the `.command.run` log in its work directory.
+    There will be a function called `nxf_launch()` that includes the command `docker run -i --cpu-shares 1024`, where `--cpu-shares` is the number of CPUs given to the process multiplied by 1024.
+
+You're probably wondering if you can set resource allocations per individual process, and the answer is of course yes, yes you can!
+We'll show you how to do that in a moment.
+
+But first, let's talk about how you can find out how much CPU and memory your processes are likely to need.
+The classic approach is to do resource profiling, meaning you run the workflow with some default allocations, record how much each process used, and from there, estimate how to adjust the base allocations.
+
+The truly excellent news on this front is that Nextflow includes built-in tools for doing this, and will happily generate a report for you on request.
+Let's try that out.
+
+### 4.3. Run the workflow to generate a resource utilization report
```

Comment:
Even during the training, we may want to run a command again. Nextflow will throw an error if we're trying to generate a new report with the same name. We can turn on override in the configuration (with `report.overwrite = true`).

Reply:
Let's see how this plays out in practice during training before deciding if we need to do this. As is, the instructions have folks running it with different filenames each time so maybe not needed.